Use latest Firefox version with Selenium 2 on GitHub Actions

The https://hub.docker.com/r/selenium/standalone-firefox image provided by Selenium only supports Firefox 47 when used with Selenium server 2.x versions. This causes problems in a test suite because a working code (e.g. dragging a div into itself in #359) fails due:

* either outdated Firefox version
* either outdated geckodriver version

Need to create an image with a modern Firefox/geckodriver version and use it on GitHub Actions.

Results with Firefox 88.0.1 and GeckoDriver 0.29.1 (actually used by Selenium 4 image: https://hub.docker.com/layers/selenium/standalone-firefox/88.0.1-geckodriver-0.29.1-20210513/images/sha256-84603a539c99ceb2ece196f0e6c4c7665b9b11bf6d48d4bd9e0015f694b50d58 ):

From Docker container logs:
```
2024-11-06 07:14:43 21:14:43.459 INFO - Launching a standalone Selenium Server
2024-11-06 07:14:43 21:14:43.502 INFO - Java: Private Build 25.292-b10
2024-11-06 07:14:43 21:14:43.502 INFO - OS: Linux 6.10.11-linuxkit amd64
2024-11-06 07:14:43 21:14:43.516 INFO - v2.53.1, with Core v2.53.1. Built from revision a36b8b1
2024-11-06 07:14:43 21:14:43.577 INFO - Driver provider org.openqa.selenium.ie.InternetExplorerDriver registration is skipped:
2024-11-06 07:14:43 registration capabilities Capabilities [{ensureCleanSession=true, browserName=internet explorer, version=, platform=WINDOWS}] does not match the current platform LINUX
2024-11-06 07:14:43 21:14:43.577 INFO - Driver provider org.openqa.selenium.edge.EdgeDriver registration is skipped:
2024-11-06 07:14:43 registration capabilities Capabilities [{browserName=MicrosoftEdge, version=, platform=WINDOWS}] does not match the current platform LINUX
2024-11-06 07:14:43 21:14:43.578 INFO - Driver class not found: com.opera.core.systems.OperaDriver
2024-11-06 07:14:43 21:14:43.578 INFO - Driver provider com.opera.core.systems.OperaDriver is not registered
2024-11-06 07:14:43 21:14:43.580 INFO - Driver provider org.openqa.selenium.safari.SafariDriver registration is skipped:
2024-11-06 07:14:43 registration capabilities Capabilities [{browserName=safari, version=, platform=MAC}] does not match the current platform LINUX
2024-11-06 07:14:43 21:14:43.581 INFO - Driver class not found: org.openqa.selenium.htmlunit.HtmlUnitDriver
2024-11-06 07:14:43 21:14:43.581 INFO - Driver provider org.openqa.selenium.htmlunit.HtmlUnitDriver is not registered
2024-11-06 07:14:43 21:14:43.648 INFO - RemoteWebDriver instances should connect to: http://127.0.0.1:4444/wd/hub
2024-11-06 07:14:43 21:14:43.649 INFO - Selenium Server is up and running
2024-11-06 07:14:51 21:14:51.796 INFO - Executing: [new session: Capabilities [{name=Behat Test, browserName=firefox, moz:firefoxOptions={args=[--headless, --width=1550, --height=1160]}}]])
2024-11-06 07:14:51 21:14:51.806 INFO - Creating a new session for Capabilities [{name=Behat Test, browserName=firefox, moz:firefoxOptions={args=[--headless, --width=1550, --height=1160]}}]
2024-11-06 07:15:37 org.openqa.selenium.firefox.NotConnectedException: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) [GFX1-]: glxtest: libpci missing
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) |[2][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.849143) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_background_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.start(NewProfileExtensionConnection.java:112)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.startClient(FirefoxDriver.java:271)
2024-11-06 07:15:37 at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:119)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:218)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:211)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:129)
2024-11-06 07:15:37 at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2024-11-06 07:15:37 at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2024-11-06 07:15:37 at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2024-11-06 07:15:37 at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor(FirefoxDriverProvider.java:91)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.FirefoxDriverProvider.newInstance(FirefoxDriverProvider.java:68)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:60)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:222)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
2024-11-06 07:15:37 at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
2024-11-06 07:15:37 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-11-06 07:15:37 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-11-06 07:15:37 at java.lang.Thread.run(Thread.java:748)
2024-11-06 07:15:37 21:15:37.435 WARN - Exception thrown
2024-11-06 07:15:37 java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
2024-11-06 07:15:37 Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
2024-11-06 07:15:37 System info: host: '0496b5b2a17a', ip: '172.17.0.2', os.name: 'Linux', os.arch: 'amd64', os.version: '6.10.11-linuxkit', java.version: '1.8.0_292'
2024-11-06 07:15:37 Driver info: driver.version: unknown
2024-11-06 07:15:37 at java.util.concurrent.FutureTask.report(FutureTask.java:122)
2024-11-06 07:15:37 at java.util.concurrent.FutureTask.get(FutureTask.java:192)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession.execute(DefaultSession.java:183)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession.<init>(DefaultSession.java:119)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession.createSession(DefaultSession.java:95)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultDriverSessions.newSession(DefaultDriverSessions.java:124)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:59)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:1)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.rest.ResultConfig.handle(ResultConfig.java:111)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.JsonHttpCommandHandler.handleRequest(JsonHttpCommandHandler.java:79)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DriverServlet.handleRequest(DriverServlet.java:204)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DriverServlet.doPost(DriverServlet.java:166)
2024-11-06 07:15:37 at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DriverServlet.service(DriverServlet.java:132)
2024-11-06 07:15:37 at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
2024-11-06 07:15:37 at org.openqa.jetty.jetty.servlet.ServletHolder.handle(ServletHolder.java:428)
2024-11-06 07:15:37 at org.openqa.jetty.jetty.servlet.ServletHandler.dispatch(ServletHandler.java:680)
2024-11-06 07:15:37 at org.openqa.jetty.jetty.servlet.ServletHandler.handle(ServletHandler.java:571)
2024-11-06 07:15:37 at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1526)
2024-11-06 07:15:37 at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1479)
2024-11-06 07:15:37 at org.openqa.jetty.http.HttpServer.service(HttpServer.java:920)
2024-11-06 07:15:37 at org.openqa.jetty.http.HttpConnection.service(HttpConnection.java:820)
2024-11-06 07:15:37 at org.openqa.jetty.http.HttpConnection.handleNext(HttpConnection.java:986)
2024-11-06 07:15:37 at org.openqa.jetty.http.HttpConnection.handle(HttpConnection.java:837)
2024-11-06 07:15:37 at org.openqa.jetty.http.SocketListener.handleConnection(SocketListener.java:243)
2024-11-06 07:15:37 at org.openqa.jetty.util.ThreadedServer.handle(ThreadedServer.java:358)
2024-11-06 07:15:37 at org.openqa.jetty.util.ThreadPool$PoolThread.run(ThreadPool.java:537)
2024-11-06 07:15:37 Caused by: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
2024-11-06 07:15:37 Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
2024-11-06 07:15:37 System info: host: '0496b5b2a17a', ip: '172.17.0.2', os.name: 'Linux', os.arch: 'amd64', os.version: '6.10.11-linuxkit', java.version: '1.8.0_292'
2024-11-06 07:15:37 Driver info: driver.version: unknown
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor(FirefoxDriverProvider.java:101)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.FirefoxDriverProvider.newInstance(FirefoxDriverProvider.java:68)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:60)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:222)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
2024-11-06 07:15:37 at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
2024-11-06 07:15:37 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-11-06 07:15:37 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-11-06 07:15:37 at java.lang.Thread.run(Thread.java:748)
2024-11-06 07:15:37 Caused by: java.lang.reflect.InvocationTargetException
2024-11-06 07:15:37 at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2024-11-06 07:15:37 at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2024-11-06 07:15:37 at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2024-11-06 07:15:37 at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
2024-11-06 07:15:37 at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor(FirefoxDriverProvider.java:91)
2024-11-06 07:15:37 ... 9 more
2024-11-06 07:15:37 Caused by: org.openqa.selenium.WebDriverException: Failed to connect to binary FirefoxBinary(/usr/bin/firefox) on port 7055; process output follows: 
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) [GFX1-]: glxtest: libpci missing
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) |[2][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.849143) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_background_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
2024-11-06 07:15:37 System info: host: '0496b5b2a17a', ip: '172.17.0.2', os.name: 'Linux', os.arch: 'amd64', os.version: '6.10.11-linuxkit', java.version: '1.8.0_292'
2024-11-06 07:15:37 Driver info: driver.version: FirefoxDriver
2024-11-06 07:15:37 at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.start(NewProfileExtensionConnection.java:124)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.startClient(FirefoxDriver.java:271)
2024-11-06 07:15:37 at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:119)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:218)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:211)
2024-11-06 07:15:37 at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:129)
2024-11-06 07:15:37 ... 14 more
2024-11-06 07:15:37 Caused by: org.openqa.selenium.firefox.NotConnectedException: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) [GFX1-]: glxtest: libpci missing
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) |[2][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.849143) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_background_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.start(NewProfileExtensionConnection.java:112)
2024-11-06 07:15:37 ... 19 more
2024-11-06 07:15:37 21:15:37.445 WARN - Exception: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) [GFX1-]: glxtest: libpci missing
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) |[2][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.849143) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_background_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
2024-11-06 07:15:37 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
2024-11-06 07:15:37 
```

From Mink:
```

Behat\Mink\Exception\DriverException : Could not open connection: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) [GFX1-]: glxtest: libpci missing
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) |[2][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.849143) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName

(firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_background_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

(firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

 /Volumes/web/g/minkphp/MinkSelenium2Driver/src/Selenium2Driver.php:347
 /Volumes/web/g/minkphp/MinkSelenium2Driver/vendor/behat/mink/src/Session.php:77
 /Volumes/web/g/minkphp/MinkSelenium2Driver/vendor/behat/mink/src/Session.php:172
 /Volumes/web/g/minkphp/MinkSelenium2Driver/vendor/mink/driver-testsuite/tests/Js/JavascriptTest.php:36
 
 Caused by
 WebDriver\Exception\UnknownError: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) [GFX1-]: glxtest: libpci missing
 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=0.847796) |[1][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.848925) |[2][GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName (t=0.849143) [GFX1-]: glxtest: libEGL missing eglGetDisplayDriverName
 
 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_background_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_color: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_widget_get_style_context: assertion 'GTK_IS_WIDGET (widget)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_state: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed
 
 (firefox:82): Gtk-CRITICAL **: gtk_style_context_get_property: assertion 'GTK_IS_STYLE_CONTEXT (context)' failed

```

I've also tried using `3.141.59-zirconium` tag and replaced Selenium Server version used. The commands are as follows:

```
git clone git@github.com:SeleniumHQ/docker-selenium.git
cd docker-selenium
git checkout refs/tags/3.141.59-zirconium

# in the "Base/Dockerfile" replaced "https://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar" with "https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.1.jar"

# in "Standalone/start-selenium-standalone.sh" file replace the

java ${JAVA_OPTS} -cp ${JAVA_CLASSPATH:-"/opt/selenium/*:."} org.openqa.grid.selenium.GridLauncherV3 \
    ${SE_OPTS}

with

java ${JAVA_OPTS} -jar /opt/selenium/selenium-server-standalone.jar \
    ${SE_OPTS}

# in "NodeFirefox/Dockerfile.txt" replace "0.26.1" to "0.35.0"
```

This gives a bit different error, but problem is the same (`glxtest: libpci missing` and `glxtest: libEGL missing`) :

```
2024-11-06 08:04:09 org.openqa.selenium.firefox.NotConnectedException: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
2024-11-06 08:04:09 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) [GFX1-]: glxtest: libpci missing
2024-11-06 08:04:09 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) |[1][GFX1-]: glxtest: libEGL missing (t=1.09112) [GFX1-]: glxtest: libEGL missing
2024-11-06 08:04:09 [Parent 101, Main Thread] WARNING: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:09 : 'glib warning', file /builds/worker/checkouts/gecko/toolkit/xre/nsSigHandlers.cpp:187
2024-11-06 08:04:09 
2024-11-06 08:04:09 ** (firefox:101): WARNING **: 06:03:26.391: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:09 
2024-11-06 08:04:09 
2024-11-06 08:04:09     at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.start(NewProfileExtensionConnection.java:112)
2024-11-06 08:04:09     at org.openqa.selenium.firefox.FirefoxDriver.startClient(FirefoxDriver.java:271)
2024-11-06 08:04:09     at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:119)
2024-11-06 08:04:09     at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:218)
2024-11-06 08:04:09     at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:211)
2024-11-06 08:04:09     at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:129)
2024-11-06 08:04:09     at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2024-11-06 08:04:09     at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2024-11-06 08:04:09     at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2024-11-06 08:04:09     at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
2024-11-06 08:04:09     at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor(FirefoxDriverProvider.java:91)
2024-11-06 08:04:09     at org.openqa.selenium.remote.server.FirefoxDriverProvider.newInstance(FirefoxDriverProvider.java:68)
2024-11-06 08:04:09     at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:60)
2024-11-06 08:04:09     at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:222)
2024-11-06 08:04:09     at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
2024-11-06 08:04:09     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2024-11-06 08:04:09     at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
2024-11-06 08:04:09     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-11-06 08:04:09     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-11-06 08:04:09     at java.lang.Thread.run(Thread.java:750)
2024-11-06 08:04:10 06:04:10.076 WARN - Exception thrown
2024-11-06 08:04:10 java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
2024-11-06 08:04:10 Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
2024-11-06 08:04:10 System info: host: '19b488d499a7', ip: '172.17.0.2', os.name: 'Linux', os.arch: 'amd64', os.version: '6.10.11-linuxkit', java.version: '1.8.0_362'
2024-11-06 08:04:10 Driver info: driver.version: unknown
2024-11-06 08:04:10     at java.util.concurrent.FutureTask.report(FutureTask.java:122)
2024-11-06 08:04:10     at java.util.concurrent.FutureTask.get(FutureTask.java:192)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultSession.execute(DefaultSession.java:183)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultSession.<init>(DefaultSession.java:119)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultSession.createSession(DefaultSession.java:95)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultDriverSessions.newSession(DefaultDriverSessions.java:124)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:59)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:1)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.rest.ResultConfig.handle(ResultConfig.java:111)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.JsonHttpCommandHandler.handleRequest(JsonHttpCommandHandler.java:79)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DriverServlet.handleRequest(DriverServlet.java:204)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DriverServlet.doPost(DriverServlet.java:166)
2024-11-06 08:04:10     at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DriverServlet.service(DriverServlet.java:132)
2024-11-06 08:04:10     at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
2024-11-06 08:04:10     at org.openqa.jetty.jetty.servlet.ServletHolder.handle(ServletHolder.java:428)
2024-11-06 08:04:10     at org.openqa.jetty.jetty.servlet.ServletHandler.dispatch(ServletHandler.java:680)
2024-11-06 08:04:10     at org.openqa.jetty.jetty.servlet.ServletHandler.handle(ServletHandler.java:571)
2024-11-06 08:04:10     at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1526)
2024-11-06 08:04:10     at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1479)
2024-11-06 08:04:10     at org.openqa.jetty.http.HttpServer.service(HttpServer.java:920)
2024-11-06 08:04:10     at org.openqa.jetty.http.HttpConnection.service(HttpConnection.java:820)
2024-11-06 08:04:10     at org.openqa.jetty.http.HttpConnection.handleNext(HttpConnection.java:986)
2024-11-06 08:04:10     at org.openqa.jetty.http.HttpConnection.handle(HttpConnection.java:837)
2024-11-06 08:04:10     at org.openqa.jetty.http.SocketListener.handleConnection(SocketListener.java:243)
2024-11-06 08:04:10     at org.openqa.jetty.util.ThreadedServer.handle(ThreadedServer.java:358)
2024-11-06 08:04:10     at org.openqa.jetty.util.ThreadPool$PoolThread.run(ThreadPool.java:537)
2024-11-06 08:04:10 Caused by: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
2024-11-06 08:04:10 Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
2024-11-06 08:04:10 System info: host: '19b488d499a7', ip: '172.17.0.2', os.name: 'Linux', os.arch: 'amd64', os.version: '6.10.11-linuxkit', java.version: '1.8.0_362'
2024-11-06 08:04:10 Driver info: driver.version: unknown
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor(FirefoxDriverProvider.java:101)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.FirefoxDriverProvider.newInstance(FirefoxDriverProvider.java:68)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:60)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:222)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
2024-11-06 08:04:10     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
2024-11-06 08:04:10     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-11-06 08:04:10     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-11-06 08:04:10     at java.lang.Thread.run(Thread.java:750)
2024-11-06 08:04:10 Caused by: java.lang.reflect.InvocationTargetException
2024-11-06 08:04:10     at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2024-11-06 08:04:10     at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2024-11-06 08:04:10     at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2024-11-06 08:04:10     at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
2024-11-06 08:04:10     at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor(FirefoxDriverProvider.java:91)
2024-11-06 08:04:10     ... 9 more
2024-11-06 08:04:10 Caused by: org.openqa.selenium.WebDriverException: Failed to connect to binary FirefoxBinary(/usr/bin/firefox) on port 7055; process output follows: 
2024-11-06 08:04:10 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) [GFX1-]: glxtest: libpci missing
2024-11-06 08:04:10 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) |[1][GFX1-]: glxtest: libEGL missing (t=1.09112) [GFX1-]: glxtest: libEGL missing
2024-11-06 08:04:10 [Parent 101, Main Thread] WARNING: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:10 : 'glib warning', file /builds/worker/checkouts/gecko/toolkit/xre/nsSigHandlers.cpp:187
2024-11-06 08:04:10 
2024-11-06 08:04:10 ** (firefox:101): WARNING **: 06:03:26.391: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:10 
2024-11-06 08:04:10 
2024-11-06 08:04:10 Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
2024-11-06 08:04:10 System info: host: '19b488d499a7', ip: '172.17.0.2', os.name: 'Linux', os.arch: 'amd64', os.version: '6.10.11-linuxkit', java.version: '1.8.0_362'
2024-11-06 08:04:10 Driver info: driver.version: FirefoxDriver
2024-11-06 08:04:10     at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.start(NewProfileExtensionConnection.java:124)
2024-11-06 08:04:10     at org.openqa.selenium.firefox.FirefoxDriver.startClient(FirefoxDriver.java:271)
2024-11-06 08:04:10     at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:119)
2024-11-06 08:04:10     at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:218)
2024-11-06 08:04:10     at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:211)
2024-11-06 08:04:10     at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:129)
2024-11-06 08:04:10     ... 14 more
2024-11-06 08:04:10 Caused by: org.openqa.selenium.firefox.NotConnectedException: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
2024-11-06 08:04:10 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) [GFX1-]: glxtest: libpci missing
2024-11-06 08:04:10 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) |[1][GFX1-]: glxtest: libEGL missing (t=1.09112) [GFX1-]: glxtest: libEGL missing
2024-11-06 08:04:10 [Parent 101, Main Thread] WARNING: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:10 : 'glib warning', file /builds/worker/checkouts/gecko/toolkit/xre/nsSigHandlers.cpp:187
2024-11-06 08:04:10 
2024-11-06 08:04:10 ** (firefox:101): WARNING **: 06:03:26.391: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:10 
2024-11-06 08:04:10 
2024-11-06 08:04:10     at org.openqa.selenium.firefox.internal.NewProfileExtensionConnection.start(NewProfileExtensionConnection.java:112)
2024-11-06 08:04:10     ... 19 more
2024-11-06 08:04:10 06:04:10.087 WARN - Exception: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
2024-11-06 08:04:10 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) [GFX1-]: glxtest: libpci missing
2024-11-06 08:04:10 Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libpci missing (t=1.09012) |[1][GFX1-]: glxtest: libEGL missing (t=1.09112) [GFX1-]: glxtest: libEGL missing
2024-11-06 08:04:10 [Parent 101, Main Thread] WARNING: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:10 : 'glib warning', file /builds/worker/checkouts/gecko/toolkit/xre/nsSigHandlers.cpp:187
2024-11-06 08:04:10 
2024-11-06 08:04:10 ** (firefox:101): WARNING **: 06:03:26.391: Failed to create DBus proxy for org.a11y.Bus: Address element “/dev/null” does not contain a colon (:)
2024-11-06 08:04:10 
2024-11-06 08:04:10 

```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -84,6 +84,9 @@ jobs:
 
       - name: Run tests
         env:
+          SELENIUM_VERSION: ${{ matrix.selenium_version }}
+          DRIVER_URL: http://localhost:4444/wd/hub
+          WEB_FIXTURES_BROWSER: firefox
           DRIVER_MACHINE_BASE_PATH: /fixtures/
         run: |
           vendor/bin/phpunit -v --coverage-clover=coverage.xml

--- a/tests/Selenium2Config.php
+++ b/tests/Selenium2Config.php
@@ -62,6 +62,18 @@ class Selenium2Config extends AbstractConfig
             return 'Checking status code is not supported.';
         }
 
+        if (
+            'Behat\Mink\Tests\Driver\Js\JavascriptTest' === $testCase
+            && 'testDragDropOntoHiddenItself' === $test
+        ) {
+            $seleniumVersion = $_SERVER['SELENIUM_VERSION'] ?? null;
+            $browser = $_SERVER['WEB_FIXTURES_BROWSER'] ?? null;
+
+            if ($seleniumVersion && version_compare($seleniumVersion, '3.0.0', '<') && $browser === 'firefox') {
+                return 'The Firefox browser compatible with Selenium Server 2.x doesn\'t fully implement drag-n-drop support.';
+            }
+        }
+
         return parent::skipMessage($testCase, $test);
     }
 

--- a/tests/Selenium2Config.php
+++ b/tests/Selenium2Config.php
@@ -7,6 +7,7 @@ use Behat\Mink\Driver\Selenium2Driver;
 use Behat\Mink\Tests\Driver\Basic\BasicAuthTest;
 use Behat\Mink\Tests\Driver\Basic\HeaderTest;
 use Behat\Mink\Tests\Driver\Basic\StatusCodeTest;
+use Behat\Mink\Tests\Driver\Js\JavascriptTest;
 
 class Selenium2Config extends AbstractConfig
 {
@@ -62,10 +63,7 @@ class Selenium2Config extends AbstractConfig
             return 'Checking status code is not supported.';
         }
 
-        if (
-            'Behat\Mink\Tests\Driver\Js\JavascriptTest' === $testCase
-            && 'testDragDropOntoHiddenItself' === $test
-        ) {
+        if (array(JavascriptTest::class, 'testDragDropOntoHiddenItself') === array($testCase, $test)) {
             $seleniumVersion = $_SERVER['SELENIUM_VERSION'] ?? null;
             $browser = $_SERVER['WEB_FIXTURES_BROWSER'] ?? null;
 

